src/test/test_marshal.cpp

```
@@ -263,38 +263,39 @@ static void compare_unmarshal_marshal(const ebpf_inst& ins1, const ebpf_inst& in
     program_info info{.platform = &platform, .type = platform.get_program_type("unspec", "unspec")};
     InstructionSeq parsed = std::get<InstructionSeq>(unmarshal(raw_program{"", "", 0, "", marshal(ins, 0), info}));
     REQUIRE(parsed.size() == 1);
     auto [_, single, _2] = parsed.back();
     (void)_;  // unused
     (void)_2; // unused
-    REQUIRE(single == ins);
+    REQUIRE(single.cmd == ins);
 }
 
-static void check_marshal_unmarshal_fail(const Instruction& ins, std::string expected_error_message,
+static void check_marshal_unmarshal_fail(const Command& ins, std::string expected_error_message,
```

cppcheck

[performance] Function parameter 'expected_error_message' should be passed by const reference.

(passedByValue)